app/viewmodels/yourAccount/ReturnsViewModel.scala

```diff
@@ -17,34 +17,74 @@
 package viewmodels.yourAccount
 
 import models.StandardPeriod
-import models.SubmissionStatus.{Due, Next, Overdue}
 import pages.{EmptyWaypoints, Waypoints}
 import play.api.i18n.Messages
 import viewmodels.{LinkModel, Paragraph, ParagraphSimple, ParagraphWithId}
 
 import java.time.format.DateTimeFormatter
 
 case class ReturnsViewModel(
-                           contents: Seq[Paragraph],
-                           linkToStart: Option[LinkModel] = None
+                             contents: Seq[Paragraph],
+                             linkToStart: Option[LinkModel] = None
                            )
 
 object ReturnsViewModel {
 
   def apply(returns: Seq[Return])(implicit messages: Messages): ReturnsViewModel = {
```

Comment:
Moved this logic to NextReturnCalculation so the filter and this model have the same basis for the next return.